Makefile

```diff
@@ -55,9 +50,10 @@ test: license-check gofmt
 	# Run the functional tests
 	make e2e-test
 
-e2e-test: bash_unit dist/flanneld-$(TAG)-$(ARCH).docker
-	./bash_unit dist/functional-test.sh
-	./bash_unit dist/functional-test-k8s.sh
+e2e-test: bash_unit dist/flanneld-e2e-$(TAG)-$(ARCH).docker
```

Comment:
Shouldn't this depend on `dist/$(REGISTRY):$(TAG)-$(ARCH).docker`

Reply:
The difference between these 2 targets are:
dist/flanneld-$(TAG)-$(ARCH).docker uses `gcr.io/google_containers/kube-cross` which is not a multiarch image and can be run only on `amd64` platform. And if I run this docker image on other than amd64 platform I'll hit exec format error.

and dist/flanneld-e2e-$(TAG)-$(ARCH).docker target uses `golang:1.8.3` docker image which is multiarch image which will load the respective docker image while running on different architecture.

Reply:
Aha, that all makes sense.